### DISCUSSION TITLE
Spikes on generated terrain

### DISCUSSION THREAD

**Discussion by laijackylai** [visgl/deck.gl#6746] [Q&A] [answered]
I am generating the terrain from custom constructed png height maps using the terrrium decoder
`// * terrarium decoder elevationDecoder: { rScaler: 256, gScaler: 1, bScaler: 1 / 256, offset: -32768 }, `
the sample input image looks like this
![6688-3576-13](https://user-images.githubusercontent.com/66239914/157045043-4f293432-b78f-497d-9eb7-db05253e9d3a.png)
along the edges seen in the above image, spikes can be seen below in the generated terrain
anyway to fix it? or is it the problem with my input image?
![Screenshot 2022-03-07 at 9 33 41 PM](https://user-images.githubusercontent.com/66239914/157044317-751c050d-1037-4a6d-8c6b-7696bff9f730.png)
![Screenshot 2022-03-07 at 9 33 44 PM](https://user-images.githubusercontent.com/66239914/157044306-88f6a054-5259-425e-ad50-b1e6750fe202.png)
![Screenshot 2022-03-07 at 9 33 52 PM](https://user-images.githubusercontent.com/66239914/157044271-49347dbf-2c9a-4065-bcb7-babb3bcf5ca8.png)

**Comment by kylebarron** [comment]
> along the edges seen in the above image, spikes can be seen below in the generated terrain

That leads me to guess that the encoding of height values into RGB might have some issue? In general I'm inclined to say this looks like a data issue.

**Comment by laijackylai** [comment, marked as answer]
yeah it was the problem with the input image, I picked the wrong algorithm while resizing the image which lead to the spikes. Thanks.
